gnat2goto/driver/tree_walk.adb

```diff
@@ -4258,7 +4258,21 @@ package body Tree_Walk is
    -----------------------
 
    function Do_Type_Reference (E : Entity_Id) return Irep is
-      (Make_Symbol_Type (Identifier => Unique_Name (E)));
+   begin
+      declare
+         Sym_Id : constant Symbol_Id := Intern (Unique_Name (E));
+      begin
+         if not Global_Symbol_Table.Contains (Sym_Id) and Is_Itype (E) then
+            Declare_Itype (E);
+         end if;
+         if not Global_Symbol_Table.Contains (Sym_Id) then
```

Comment:
If the "checking the same condition twice" is intentional, can we have a comment to stop people refactoring it?  If no, can we refactor it now?